### PR TITLE
Feature/#582 unit visibility toggle 2

### LINKED ISSUE #582: Add visible flag to unit / Make available on
User Story

### As a: 

Teacher

### I want:

to decide for every unit in a course if these is visible or not. A possible use case could be: setup the course at the beginning of the semester but hide the content. 

### so that:

Unit is visible when checkbox is on. By default the checkbox should be checked. When "unit is not visible" checked than there should be another option "make available on" with a datepicker.  When current date is greater/equal to date input the visible flag should be changed to visible.

## Acceptance criteria:
- [x] Checkbox or Toggle: Unit visible / Unit not visible
- [ ] Datepicker whe Unit not visible
- [ ] Unit is visible when current date > input date

## Additional info:

This issue is blocked by #556

### PR DESCRIPTION
**Pull request by dboschm**
## Description:
Simple Visiblility Toggle for Units (without Datepicker/visibleFromDate)

Closes #582

## Improvements
- Added visibility flag to IUnit
- Added toggle to general info form 
- Added match filter to courseController, to only populate visible units and skip invisible if currentUser is not teacher or admin

## Known Issues:
- visibleFromDate has not been implemented after consultation to @danielkesselberg. This raises some strage questions like: which date should be taken if teacher does not want to publish the unit and does not know when. 
- Unit forms dataflow has to be refactored/unified so that sub components does not know anything about supercomponents. 

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: api/src/controllers/CourseController.ts [hunk 1]
```diff
@@ -234,11 +234,14 @@ export class CourseController {
...
     }
 
+    const isTeacherOrAdmin= (currentUser.role === 'teacher' || currentUser.role === 'admin');
+
     await course.populate({
         path: 'lectures',
         populate: {
           path: 'units',
           virtuals: true,
+          match: {$or: [{visible: undefined},{visible: true},{visible: !isTeacherOrAdmin}]},
```
Comment (danielkesselberg) [inline api/src/controllers/CourseController.ts]: Nice one!

**Comment by dboschm** [comment]
Sorry, package.json and package-lock.json have been commited and pushed accidentally. Please do not merge.

**Comment by PatrickSkowronek** [comment]
There are many Problems, we should discuss these Tomorrow.

**Comment by dboschm** [comment]
Ok. I am not able to reproduce this error. Could you show me the error in a personal session.

**Pull request merged**